### PR TITLE
refactor(backend): simplify pyarrow batching code

### PR DESCRIPTION
**Pull request by cpcloud**
This PR simplifies a bit of the code around getting schemas for pyarrow batching, and adds a new `Schema.as_struct` method to convert `Schema` objects into `Struct` types.

### CODE REVIEW

**Comment by cpcloud** [comment]
Patch coverage is failing because we don't include `pyarrow` as dependency of the clickhouse backend, so here's a test run of the `to_pyarrow*`-related tests:

```
…/ibis on  simplify-pyarrow-batching is 📦 v4.0.0 via 🐍 v3.10.9 via ❄️  impure (ibis-3.10.9) took 11s
❯ pytest -m clickhouse ibis/backends/tests/test_export.py -q
.................                                                                                                                                                                              [100%]
17 passed, 238 deselected in 0.53s
```

File: ibis/expr/schema.py [hunk 1]
```diff
@@ -215,6 +215,9 @@ def to_pyarrow(self):
 
         return ibis_to_pyarrow_schema(self)
 
+    def as_struct(self) -> dt.Struct:
```
Comment (kszucs) [inline ibis/expr/schema.py:218]: +1

File: ibis/backends/base/__init__.py [hunk 2]
```diff
@@ -275,7 +263,7 @@ def to_pyarrow(
         except ValueError:
             # The pyarrow batches iterator is empty so pass in an empty
             # iterator and a pyarrow schema
-            schema = self._table_or_column_schema(expr)
+            schema = expr.as_table().schema()
```
Comment (kszucs) [inline ibis/backends/base/__init__.py:266]: +1

**Review by kszucs — APPROVED** [review]

**Comment by kszucs** [comment]
@cpcloud can you please add scope to the commit message?

**Comment by cpcloud** [comment]
@kszucs I decided to squash instead. Setting to automerge.

**Pull request merged**